### PR TITLE
creates a booking

### PR DESCRIPTION
**Pull request by olliebrownlow**
Co-authored-by: VladC24 <vlad.chiriac24@gmail.com>
Co-authored-by: Oliver Brownlow <olliebrownlow@gmail.com>

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -61,12 +61,19 @@ class MakersBNB < Sinatra::Base
 
   get '/spaces/bookings/:id' do
     id = params[:id]
+    p id
```
Comment (breycarr) [inline app.rb]: Does this still need to print at the commit stage?

File: app.rb [hunk 2]
```diff
@@ -61,12 +61,19 @@ class MakersBNB < Sinatra::Base
...
 
-  # get '/spaces/requests/:id' do
-  #   erb :'spaces/requests'
-  # end
+  post '/spaces/request' do
+    space_id = session[:id]
+    # flash messsage 'Request to book has been sent'
+    booking = Booking.create(arrival: params[:request_from], departure: params[:request_until], space_id: space_id)
+    p booking
+    p booking.saved?
```
Comment (breycarr) [inline app.rb]: Do these still need to print at the commit stage?

File: lib/booking.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Booking
+  include DataMapper::Resource
+
+  property :id,           Serial
+  property :arrival,      Date,     required: true
+  property :departure,    Date,     required: true
+  property :confirmed,    Boolean
+
+  belongs_to :space
+
```
Comment (breycarr) [inline lib/booking.rb]: Does this need to belong_to :user as well?

File: lib/user.rb [hunk 4]
```diff
@@ -9,6 +9,7 @@ class User
 
   has n, :spaces
 
+
```
Comment (breycarr) [inline lib/user.rb]: Does this extra line need to be here?

File: spec/helpers/list_space.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+def list_space
```
Comment (breycarr) [inline spec/helpers/list_space.rb]: I think we've created additional helpers in the helpers folder, which may cause some issues when this merges.

File: app.rb (hunk 1 above)
Comment (olliebrownlow) [reply app.rb]: Will change before we merge.

File: app.rb (hunk 2 above)
Comment (olliebrownlow) [reply app.rb]: see above

File: lib/booking.rb (hunk 3 above)
Comment (olliebrownlow) [reply lib/booking.rb]: Yes! I think it does

File: lib/user.rb (hunk 4 above)
Comment (olliebrownlow) [reply lib/user.rb]: Will change before we merge.

File: spec/helpers/list_space.rb (hunk 5 above)
Comment (olliebrownlow) [reply spec/helpers/list_space.rb]: Thanks for the heads up!

**Pull request merged**